### ISSUE TITLE
reference-section-title - Not inserted when documents end with a header

### ISSUE THREAD

**Issue by odkr** [jgm/pandoc-citeproc#431]
The manual states that:

> If [reference-section-title] has a value, a section header with this title will be
> added before the bibliography. If reference-section-title is
> not specified and the document ends with a section header,  this
> final header will be treated as the bibliography header.

However, even if `reference-section-title` has a value, as long as the document ends with a section header, no additional header is inserted.

Compare:

```sh
$ pandoc -F pandoc-citeproc <<EOF
> ---
> reference-section-title: My references
> references:
> - id: doe
>   type: book
>   author:
>   - family: Doe
>     given: Jane
>   issued:
>   - year: 1000
>   title: A theory of everything
> ---
>
> @doe says p!
> EOF
<p><span class="citation" data-cites="doe">Doe (1000)</span> says p!</p>
<h1 id="bibliography" class="unnumbered">My references</h1>
<div id="refs" class="references" role="doc-bibliography">
<div id="ref-doe">
<p>Doe, Jane. 1000. <em>A Theory of Everything</em>.</p>
</div>
</div>
```

```sh
$ pandoc -F pandoc-citeproc <<EOF
> ---
> reference-section-title: My references
> references:
> - id: doe
>   type: book
>   author:
>   - family: Doe
>     given: Jane
>   issued:
>   - year: 1000
>   title: A theory of everything
> ---
>
> @doe says p!
>
> # Not a bibliography header
>
> EOF
<p><span class="citation" data-cites="doe">Doe (1000)</span> says p!</p>
<h1 id="not-a-bibliography-header" class="unnumbered">Not a bibliography header</h1>
<div id="refs" class="references" role="doc-bibliography">
<div id="ref-doe">
<p>Doe, Jane. 1000. <em>A Theory of Everything</em>.</p>
</div>
</div>
```

If I could add a "low priority" tag, I would. But I felt I should point out the documentation and the actual behaviour don’t match.

**Comment by jgm** [comment]
For reference, here's what the docs say:
```
`reference-section-title`
:    If this has a value, a section header with this title will be
     added before the bibliography.  If `reference-section-title`
     is not specified and the document ends with a section header,
     this final header will be treated as the bibliography header.
     Note: `reference-section-title` has no effect when the bibliography
     is placed explicitly using a Div with id `refs`.
```
Suggested alternative:
```
`reference-section-title`
:    Specifies a fallback title for the bibliography, for the case where
    the document does not end with a section heading and no Div
    with id `refs` is used to place the bibliography explicitly.  If these
    conditions obtain and `reference-section-title` is not set, the
    bibliography will be included at the end of the document with no
    section heading.
```
Actually, though, it would make more sense to me to include a new heading with `reference-section-title` even if the document ends with a section heading.

**Comment by odkr** [comment]
I agree that it would make more sense if `reference-section-title` always inserted a new heading. I stumbled upon this when I previewed a draft that I’m working on, and which, at that point, had a dangling section header. This sent me on a search, not a long one, for where the header of my reference section went. It wasn’t hard to guess that `reference-section-title` is really a fallback. But other people may take longer to realise what's going on, of course. So it'd be good not to send them on that search in the firstplace. But then, this is corner case. And most users of pandoc-citeproc won't have a hard time figuring out what happened. So just updating the documentation certainly is enough. Hmm, that wasn’t helpful, was it?
